web/client/components/maps/modals/__tests__/MetaDataModal-test.jsx

```diff
@@ -56,6 +57,42 @@ describe('This test for MetadataModal', () => {
         const closeBtnList = modalDivList.item(0).getElementsByTagName('button');
         expect(closeBtnList.length).toBe(2);
     });
+    /*
+     * This checks if you can close the modal even if the function (confirm) is not defined.
+     * see https://github.com/geosolutions-it/MapStore2/issues/2576
+     */
+    it('Test MetadataModal onToggleUnsavedChangesModal only if present', () => {
+        let thumbnail = "myThumnbnailUrl";
+        let errors = ["FORMAT"];
+        let map = {
+            unsavedChanges: true,
+            thumbnail: thumbnail,
+            id: 123,
+            canWrite: true,
+            errors: errors
+        };
+        const actions = {
+            onToggleUnsavedChangesModal: () => {},
+            onDisplayMetadataEdit: () => {}
+        };
+        const spyonToggleUnsavedChangesModal = expect.spyOn(actions, 'onToggleUnsavedChangesModal');
+        const spyonDisplayMetadataEdit = expect.spyOn(actions, 'onDisplayMetadataEdit');
+        const cmp = ReactDOM.render(<MetadataModal
+            show useModal map={map} id="MetadataModal"
+            detailsSheetActions={{onToggleUnsavedChangesModal: actions.onToggleUnsavedChangesModal}} onDisplayMetadataEdit={actions.spyonDisplayMetadataEdit} />, document.getElementById("container"));
+        expect(cmp).toExist();
+        let el = document.querySelector('#ms-resizable-modal .btn-group button');
+        expect(el).toExist();
+        ReactTestUtils.Simulate.click(el); // <-- trigger event callback
+        expect(spyonToggleUnsavedChangesModal).toHaveBeenCalled();
+        expect(spyonDisplayMetadataEdit).toNotHaveBeenCalled();
+        ReactDOM.render(<MetadataModal
+            show useModal map={map} id="MetadataModal"
+            onDisplayMetadataEdit={actions.onDisplayMetadataEdit} />, document.getElementById("container"));
+        el = document.querySelector('#ms-resizable-modal .btn-group button');
+        ReactTestUtils.Simulate.click(el);
```

Comment:
does it makes sense to had here 
expect(spyonToggleUnsavedChangesModal).toNotHaveBeenCalled(); ?

Reply:
no, because if it is passed it will be triggered instead of `onDisplayMetadataEdit`. See [this row](https://github.com/geosolutions-it/MapStore2/pull/2587/files/4c5000979838f872a9bf09838580be1bede8b009#diff-4b95be631fd0d83523ad56476a234ffaR167)